src/test/scala/is/hail/expr/ir/OrderingSuite.scala

```diff
@@ -35,4 +35,51 @@ class OrderingSuite {
     }
     p.check()
   }
+
+  @Test def testEqualityInEmitFunction() {
+    val compareGen = Type.genArb.flatMap(t => Gen.zip(Gen.const(t), t.genNonmissingValue, t.genNonmissingValue))
+    val p = Prop.forAll(compareGen) { case (t, a1, a2) =>
+
+      val ttup = TTuple(t)
+      val region = Region()
+      val rvb = new RegionValueBuilder(region)
+
+      rvb.start(ttup)
+      rvb.addAnnotation(ttup, Row(a1))
+      val tup1 = rvb.end()
+
+      rvb.start(ttup)
+      rvb.addAnnotation(ttup, Row(a2))
+      val tup2 = rvb.end()
+
+      val v1 = ttup.loadField(region, tup1, 0)
+      val v2 = ttup.loadField(region, tup2, 0)
+
+      val compare = t.unsafeOrdering(true).compare(region, v1, region, v2)
+
+      val ir = Apply("==", FastSeq(GetTupleElement(In(0, ttup), 0), GetTupleElement(In(1, ttup), 0)))
+      val irfb = EmitFunctionBuilder[Region, Long, Boolean, Long, Boolean, Boolean]
+
+      Infer(ir)
+      Emit(ir, irfb)
+
+      val irf = irfb.result()()
+      assert(irf(region, tup1, false, tup2, false) == (compare == 0))
+      assert(irf(region, tup1, false, tup1, false))
+      assert(irf(region, tup2, false, tup2, false))
+
+      val ir2 = Apply("!=", FastSeq(GetTupleElement(In(0, TTuple(t)), 0), GetTupleElement(In(1, TTuple(t)), 0)))
+      val irfb2 = EmitFunctionBuilder[Region, Long, Boolean, Long, Boolean, Boolean]
+
+      Infer(ir2)
+      Emit(ir2, irfb2)
+
+      val irf2 = irfb2.result()()
+      assert(irf2(region, tup1, false, tup2, false) == (compare != 0))
+      assert(!irf2(region, tup1, false, tup1, false))
+      assert(!irf2(region, tup2, false, tup2, false))
+      true
+    }
+    p.check()
+  }
```

Comment:
is findFirstIn being tested?

Reply:
the string function? those are being tested in Python.